pkg/utils/status.go

```diff
@@ -881,3 +965,104 @@ func observabilityStatusCheck(ctx context.Context, instance *csmv1.ContainerStor
 
 	return false, nil
 }
+
+// authProxyStatusCheck - calculate success state for auth proxy
+func authProxyStatusCheck(ctx context.Context, instance *csmv1.ContainerStorageModule, r ReconcileCSM, _ *csmv1.ContainerStorageModuleStatus) (bool, error) {
+	log := logger.GetLogger(ctx)
+	certEnabled := false
+	nginxEnabled := false
+
+	for _, m := range instance.Spec.Modules {
+		if m.Name == csmv1.AuthorizationServer {
+			for _, c := range m.Components {
+				if c.Name == "ingress-nginx" && *c.Enabled {
+					nginxEnabled = true
+				}
+				if c.Name == "cert-manager" && *c.Enabled {
+					certEnabled = true
+				}
+			}
+		}
+	}
+
+	opts := []client.ListOption{
+		client.InNamespace(instance.GetNamespace()),
+	}
+	deploymentList := &appsv1.DeploymentList{}
+	err := r.GetClient().List(ctx, deploymentList, opts...)
+	if err != nil {
+		return false, err
+	}
+
+	checkFn := func(deployment *appsv1.Deployment) bool {
+		return deployment.Status.ReadyReplicas == *deployment.Spec.Replicas
+	}
+
+	for _, deployment := range deploymentList.Items {
+		deployment := deployment
+		switch deployment.Name {
+		case "authorization-ingress-nginx-controller":
```

Comment:
This needs to be dynamic for the namespace, unless we are reverting that change.

Reply:
that change is coming after this PR goes in @atye